app/integration/project-generator/src/main/java/io/syndesis/integration/project/generator/ProjectGenerator.java

```diff
@@ -82,6 +98,94 @@ public InputStream generate(IntegrationDeployment deployment) throws IOException
         return is;
     }
 
+    @Override
+    public Properties generateApplicationProperties(IntegrationDeployment deployment) {
+        final Properties properties = new Properties();
+        final List<? extends Step> steps = deployment.getSpec().getSteps();
+
+        // ****************************************
+        //
+        // Connectors
+        //
+        // ****************************************
+
+        steps.stream()
+            .filter(step -> step.getStepKind().equals("endpoint"))
+            .filter(step -> step.getAction().filter(ConnectorAction.class::isInstance).isPresent())
+            .filter(step -> step.getConnection().isPresent())
+            .forEach(step -> {
+                final String index = step.getMetadata(Step.METADATA_STEP_INDEX).orElseThrow(() -> new IllegalArgumentException("Missing index for step:" + step));
+                final Connection connection = step.getConnection().get();
+                final ConnectorAction action = ConnectorAction.class.cast(step.getAction().get());
+                final ConnectorDescriptor descriptor = action.getDescriptor();
+                final Connector connector = resolveConnector(connection, resourceManager);
+
+                if (connector.getComponentScheme().isPresent() || descriptor.getComponentScheme().isPresent()) {
+                    // Grab the component scheme from the component descriptor or
+                    // from the connector
+                    final String componentScheme = Optionals.first(descriptor.getComponentScheme(), connector.getComponentScheme()).get();
+
+                    Stream.of(connector, connection, step)
+                        .filter(WithConfiguredProperties.class::isInstance)
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isSecret, action::isSecret))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s-%s.%s", componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
```

Comment:
What happens when `decrypt() failed ? Should the whole processing be stopped or only this particular secret ? Would be nice to add the key here to the error message, too.

Reply:
The encrytion component does not explicit throws exceptions but return null if fails, this can of course be checked (see https://github.com/syndesisio/syndesis/blob/master/app/rest/dao/src/main/java/io/syndesis/dao/manager/EncryptionComponent.java#L102-L117)